### PR TITLE
Make output schema to match selection list for aggregation groupbys

### PR DESCRIPTION
**Pull request by fx19880617**
- Query results schema should match select list.
- Validate that non-aggregate expression in select list must be in group by expressions.
E.g. 
```
select group_id , count(*), sum(rsvp_count),count(rsvp_count), sum(rsvp_count),ADD(group_id,1)  from meetupRsvp group by  group_id , ADD(group_id,1)   limit 10
```
Results is 
```
{
    "resultTable": {
        "dataSchema": {
            "columnDataTypes": ["LONG", "LONG", "DOUBLE", "LONG", "DOUBLE", "DOUBLE"],
            "columnNames": ["group_id", "count(*)", "sum(rsvp_count)", "count(*)", "sum(rsvp_count)", "add(group_id,'1')"]
        },
        "rows": [
            [263790, 2, 2.0, 2, 2.0, 263791.0],
            [15915612, 1, 1.0, 1, 1.0, 1.5915613E7],
            [1516787, 1, 1.0, 1, 1.0, 1516788.0],
            [31461301, 1, 1.0, 1, 1.0, 3.1461302E7],
            [3315552, 2, 2.0, 2, 2.0, 3315553.0],
            [19068870, 1, 1.0, 1, 1.0, 1.9068871E7],
            [252197, 1, 1.0, 1, 1.0, 252198.0],
            [32449323, 1, 1.0, 1, 1.0, 3.2449324E7],
            [31525187, 1, 1.0, 1, 1.0, 3.1525188E7],
            [18402988, 1, 1.0, 1, 1.0, 1.8402989E7]
        ]
    },
    "exceptions": [],
    "numServersQueried": 1,
    "numServersResponded": 1,
    "numSegmentsQueried": 1,
    "numSegmentsProcessed": 1,
    "numSegmentsMatched": 1,
    "numConsumingSegmentsQueried": 1,
    "numDocsScanned": 163,
    "numEntriesScannedInFilter": 0,
    "numEntriesScannedPostFilter": 326,
    "numGroupsLimitReached": true,
    "totalDocs": 163,
    "timeUsedMs": 18,
    "segmentStatistics": [],
    "traceInfo": {},
    "minConsumingFreshnessTimeMs": 1580132510553
}
```

### CODE REVIEW

File: pinot-core/src/main/java/org/apache/pinot/core/query/reduce/GroupByDataTableReducer.java [hunk 1]
```diff
@@ -194,14 +210,104 @@ private void setSQLGroupByInResultTable(BrokerResponseNative brokerResponseNativ
...
+    if (expression.getFunctionCall() != null) {
+      // handle AS
+      if (expression.getFunctionCall().getOperator().equalsIgnoreCase("AS")) {
+        return getExpressionMapIdx(dataSchema, expression.getFunctionCall().getOperands().get(0), nextAggregationIdx);
+      }
+      // Return next aggregation idx.
+      return nextAggregationIdx;
+    }
+    // Handle identifier, which is a column.
+    if (expression.getIdentifier() != null) {
```
Comment (npawar) [inline pinot-core/src/main/java/org/apache/pinot/core/query/reduce/GroupByDataTableReducer.java]: When will this if block be needed? These will always be in the group by clause, and hence be handled by 269 right?

File: pinot-core/src/main/java/org/apache/pinot/core/query/reduce/GroupByDataTableReducer.java [hunk 2]
```diff
@@ -194,14 +210,104 @@ private void setSQLGroupByInResultTable(BrokerResponseNative brokerResponseNativ
...
+
+  /**
+   * Trying to match an expression based on given groupByList.
+   *
+   * @param groupByList
+   * @param expression
+   * @return matched idx from groupByList
+   */
+  private int getGroupByIdx(List<Expression> groupByList, Expression expression) {
+    for (int i = 0; i < groupByList.size(); i++) {
```
Comment (npawar) [inline pinot-core/src/main/java/org/apache/pinot/core/query/reduce/GroupByDataTableReducer.java]: this group by list will be scanned for every single expression in the selection. Do you think it's worth constructing a map upfront groupByExpr -> index , within getFinalSchemaIdx method?

File: pinot-core/src/main/java/org/apache/pinot/core/query/reduce/GroupByDataTableReducer.java [hunk 3]
```diff
@@ -194,14 +210,104 @@ private void setSQLGroupByInResultTable(BrokerResponseNative brokerResponseNativ
         values[index] = _aggregationFunctions[aggNum++].extractFinalResult(values[index]);
         index++;
       }
-      rows.add(values);
+      if (_sqlSelectionList != null) {
+        Object[] finalValues = new Object[_sqlSelectionList.size()];
+        for (int i = 0; i < finalSchemaMapIdx.length; i++) {
+          if (finalSchemaMapIdx[i] == -1) {
```
Comment (npawar) [inline pinot-core/src/main/java/org/apache/pinot/core/query/reduce/GroupByDataTableReducer.java]: should this ever happen? the validation during parsing should prevent this right?

**Comment by npawar** [comment]
Thanks for fixing this! I just have minor questions. LGTM otherwise

File: pinot-core/src/main/java/org/apache/pinot/core/query/reduce/GroupByDataTableReducer.java (hunk 1 above)
Comment (fx19880617) [reply pinot-core/src/main/java/org/apache/pinot/core/query/reduce/GroupByDataTableReducer.java]: Right, will remove this block

File: pinot-core/src/main/java/org/apache/pinot/core/query/reduce/GroupByDataTableReducer.java (hunk 3 above)
Comment (fx19880617) [reply pinot-core/src/main/java/org/apache/pinot/core/query/reduce/GroupByDataTableReducer.java]: true, removed this block

File: pinot-core/src/main/java/org/apache/pinot/core/query/reduce/GroupByDataTableReducer.java (hunk 2 above)
Comment (fx19880617) [reply pinot-core/src/main/java/org/apache/pinot/core/query/reduce/GroupByDataTableReducer.java]: I'm not too worried about this as the check would be `selectionSize + groupBySize * groupBySize` times

**Pull request merged**